lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py

```diff
@@ -374,6 +374,15 @@ def wait_for_event(self, filter=None, timeout=None):
             )
         return None
 
+    def wait_for_events(self, events, timeout=None):
+        """Wait for a list of events in `events` in any order.
+        Return the events not hit before the timeout expired"""
+        events = events[:]  # Make a copy to avoid modifying the input
+        while events:
+            event = self.wait_for_event(filter=events, timeout=timeout)
+            events.remove(event["event"])
```

Comment:
If we hit a timeout, wont `event` be `None`?

Reply:
Ah yeah it looks like its not handling the timeout correctly.